sdk/core/CMakeLists.txt

```
@@ -0,0 +1,12 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required (VERSION 3.12)
```

Make it consistent with other cmakelists and docs.
https://github.com/Azure/azure-sdk-for-cpp/blob/f285aaead94be16bd3a7af8ac6565aeb4d55a525/CONTRIBUTING.md#cmake
```suggestion
cmake_minimum_required (VERSION 3.13)
```

Identity is the only one that is inconsistent, which we should update as well:
https://github.com/Azure/azure-sdk-for-cpp/blob/a38de03e7b42cdf1f6df61dae86127a9bc4554e7/sdk/identity/CMakeLists.txt#L4